ambry-router/src/main/java/com.github.ambry.router/NonBlockingRouter.java

```diff
@@ -215,7 +215,30 @@ public void onCompletion(GetBlobResultInternal internalResult, Exception excepti
   @Override
   public Future<String> stitchBlob(BlobProperties blobProperties, byte[] userMetadata, List<ChunkInfo> chunksToStitch,
```

Comment:
Do we do any validation for `chunksToStitch` in the router? Is that done outside of it? (ensuring they are in order and complete). Wondering if it makes sense to embed any stitch-related information in the blob id of the chunks when they get uploaded.

Reply:
The verification of the chunks that we do is done in the `PutOperation.processChunksToStitch()`, but that verification trusts the `ChunkInfo` fields. It does not actually fetch blobs to ensure that they match the `ChunkInfo` fields.

The frontend logic builds `ChunkInfo` by parsing and verifying a signed ID, which, in a secure implementation, should contain the blob ID, the additional stitch-related info, and a signature to prevent corruption.